### ISSUE TITLE
Cannot hover data point under a brush

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#3154]
For example, given a new example I add for https://github.com/vega/vega-lite/pull/3146, I cannot trigger a brushed point to be blue even though the condition for hover comes first. 

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "description": "Drag out a rectangular brush to highlight points.",
  "data": {"url": "data/cars.json"},
  "selection": {
    "brush": {
      "type": "interval"
    }, 
    "hoverbrush": {
      "type": "multi", "on": "mouseover",
      "nearest": true,
      "empty": "none"
    }
  },
  "mark": "point",
  "encoding": {
    "x": {"field": "Horsepower", "type": "quantitative"},
    "y": {"field": "Miles_per_Gallon", "type": "quantitative"},
    "color": {
      "condition": [ 
        {"selection": "hoverbrush", "value": "blue"},
        {"selection": "brush", "value": "skyblue"}
      ],
      "value": "grey"
    }
  }
} 
```

A more important issue would be that even with just one condition and brush (e.g., the `brush.vl.json`), we cannot hover a data point to see tooltip.

**Comment by arvind** [comment]
Yes, this is a limitation of Vega's event handling (see vega/vega-scenegraph#53). Marks that render on top of others capture events and do not propagate them down.

**Comment by kanitw** [comment]
Before we close this enitrely, I wonder if it makes sense to have an option to show brush below points? The fact is that we cant hover to enable tooltip is quite hurting

**Comment by arvind** [comment]
A couple of points:

(1) I would be uncomfortable if implementation limitations impacted language level concerns. That strikes me as a case of the tail wagging the dog. 

(2) If the brush was rendered below points, that would interfere with the ability to smoothly drag the brush. If a drag event was initiated over a mark, or if a drag moved over a mark, the brush would no longer be draggable. That feels like a worse violation of user expectations than this limitation. With that said, perhaps we should explicitly document this behavior as a current limitation given that we will now support an array of conditional value definitions.

**Comment by kanitw** [comment]
> (1) I would be uncomfortable if implementation limitations impacted language level concerns. That strikes me as a case of the tail wagging the dog.

It depends on whether there is a case where users would want the brush to be under the points. 
Perhaps not, so I think this is a fair point.

> If a drag event was initiated over a mark, or if a drag moved over a mark, the brush would no longer be draggable. 

Yeah, this is definitely annoying. 

> If the brush was rendered below points, that would interfere with the ability to smoothly drag the brush. 

(Minor note) Given that the drag depends on window.mousemove, I'm not sure how is this a problem. I just tried moving the brush in the output layer in the output Vega spec and it seems fine.

> With that said, perhaps we should explicitly document this behavior as a current limitation given that we will now support an array of conditional value definitions.

Yeah, let's just document the limitations for now. This seems like a very corner case anyway.

**Comment by arvind** [comment]
> (Minor note) Given that the drag depends on window.mousemove, I'm not sure how is this a problem. I just tried moving the brush in the output layer in the output Vega spec and it seems fine.

Dragging is initiated on a transparent brush mark rendered on top of all marks. If this brush mark didn’t exist (and instead dragging was triggered by the background mark), users would experience an inconsistent drag start experience.

**Comment by kanitw** [comment]
https://github.com/vega/vega-lite/issues/5728 is another request for this.

**Comment by domoritz** [comment]
Just fyi, here is the documentation for Vega tooltip handlers: https://vega.github.io/vega/docs/api/view/#view_tooltip. Tooltips can also be implemented in Vega itself but the handler is great for html tooltips and it's also what we use for the Vega tooltip plugin.

**Comment by BorisVSchmid** [comment]
+1 on the lack of tooltips (or the ability to inspect a point) when using a brush being an important bug, rather than a niche case. An example of how I would be using the tooltips when using a brush is in the image below: the brush selects the time window, and by hovering over the points I can inspect individual points through the tooltip.

<img width="1489" alt="Screenshot 2022-04-21 at 14 46 16" src="https://user-images.githubusercontent.com/25182535/164461165-c08a66fb-5f22-49a1-a894-ea319ddaf226.png">

Outside of that, fantastic to be able to make these visualizations in <20 lines (using clojure / oz wrapper around vega-lite.

````
(oz/view! {:vconcat [{:width 1400
                      :height 500
                      :layer [{:data {:values (slurp "observables/world-110m.json") :format {:type "topojson" :feature "countries"}}
                               :projection {:scale 650 :center [25 52] :clipExtent [[0 0] [1400 500]]}
                               :mark {:type "geoshape" :fill "lightgrey" :stroke "grey"}}
                              {:data {:values plague-db}
                               :transform [{:filter {:selection "brush"}}]
                               :projection {:scale 650 :center [25 52] :clipExtent [[0 0] [1400 500]]}
                               :encoding {:longitude {:field :longitude}
                                          :latitude {:field :latitude}}
                               :mark {:type "circle" :fill "black" :tooltip {:content "data"}}}]}
                     {:width 1400
                      :height 200
                      :params [{:name "brush" :select {:type "interval" :encodings ["x"]}}]
                      :data {:values plague-db}
                      :encoding {:x {:field :year :type "quantitative"} :y {:aggregate "count"}}
                      :mark {:type "bar"}}]}
          :port 3333)
````
